packages/eslint-plugin/tests/rules/block-spacing.test.ts

```diff
@@ -0,0 +1,92 @@
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+
+import rule from '../../src/rules/block-spacing';
+import type { InvalidTestCase, ValidTestCase } from '../RuleTester';
+import { RuleTester } from '../RuleTester';
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+});
+
+type InvalidBlockSpacingTestCase = InvalidTestCase<
+  'missing' | 'extra',
+  ['always' | 'never']
+>;
+
+const options = ['always', 'never'] as const;
+const typeDeclaration = [
```

Comment:
Nit: other arrays are named as the plural. Let's keep it consistent for this one.

```suggestion
const typeDeclarations = [
```